vsphere/datadog_checks/vsphere/api.py

```diff
@@ -115,13 +115,13 @@ def smart_connect(self):
             # Remove type ignore when this is merged https://github.com/python/typeshed/pull/3855
             context = ssl.SSLContext(ssl.PROTOCOL_TLS)  # type: ignore
             context.verify_mode = ssl.CERT_NONE
-        elif self.config.ssl_capath:
+        elif self.config.ssl_cafile or self.config.ssl_capath:
             # Remove type ignore when this is merged https://github.com/python/typeshed/pull/3855
             context = ssl.SSLContext(ssl.PROTOCOL_TLS)  # type: ignore
             context.verify_mode = ssl.CERT_REQUIRED
             # `check_hostname` must be enabled as well to verify the authenticity of a cert.
             context.check_hostname = True
-            context.load_verify_locations(capath=self.config.ssl_capath)
+            context.load_verify_locations(cafile=self.config.ssl_cafile, capath=self.config.ssl_capath)
```

Comment:
Noting this aligns with how we handle this behavior in the base TLS wrapper https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/utils/tls.py#L122-L130